web/cobrands/smidsy/js/assets.js

```diff
@@ -0,0 +1,72 @@
+(function(){
+
+if (!fixmystreet.maps) {
+    return;
+}
+
+var heatmap_style = new OpenLayers.Style({}, {
+    context: {
+        getStrokeColor: function(feature) {
+            // "density" is incidents-per-metre for this road feature
+            var density = parseFloat(feature.attributes.density);
+            if (density < 0.001) {
+                return "#fff5eb";
+            } else if (density < 0.01) {
+                return "#fed2a6";
+            } else if (density < 0.1) {
+                return "#fd9243";
+            } else if (density < 1) {
+                return "#df4f05";
+            } else {
+                return "#7f2704";
+            }
+        },
+        getStrokeWidth: function() {
+            return {
+                0: 2,
+                1: 3,
+                2: 5,
+                3: 8,
+                4: 12,
+                5: 16
+            }[fixmystreet.map.zoom];
+        }
+    }
+});
+heatmap_style.addRules([new OpenLayers.Rule({
+    symbolizer: {
+        strokeColor: '${getStrokeColor}',
+        strokeLinecap: 'butt',
+        strokeWidth: '${getStrokeWidth}',
+        strokeOpacity: 0.9
+    }
+})]);
+
+
+fixmystreet.assets.add({
+    http_options: {
+        url: "https://tilma.staging.mysociety.org/mapserver/collideoscope",
+        params: {
+            SERVICE: "WFS",
+            VERSION: "1.1.0",
+            REQUEST: "GetFeature",
+            SRSNAME: "urn:ogc:def:crs:EPSG::3857",
+            TYPENAME: "heatmap"
+        }
+    },
+    format_class: OpenLayers.Format.GML.v3.MultiCurveFix,
+    asset_type: 'spot',
+    stylemap: new OpenLayers.StyleMap({
+        'default': heatmap_style
+    }),
+    max_resolution: 19.109257068634033,
+    min_resolution: 0.5971642833948135,
+    non_interactive: true,
+    geometryName: 'msGeometry',
+    srsName: "EPSG:3857",
+    strategy_class: OpenLayers.Strategy.FixMyStreet,
+    name: "Heatmap",
+    asset_category: ""
```

Comment:
There is a tiny bug here, e.g. click map to report then click back, the heatmap is then hidden but the sub map link still says "Hide heatmap". I guess you'd want to use `always_visible` here (also then doesn't call some unneeded code to do with category assets) but then it is visible at load. Could add a new var to set default visibility of always_visible assets? Feels like it should be something passed in to the constructor anyway; `visibility` is only thing that can't be overridden in fact, if I'm reading it right.

Reply:
@zarino suggested it should be visible by default, so perhaps setting `always_visible` would be OK here.